ark/utils/test_utils.py

```diff
@@ -0,0 +1,256 @@
+    label_data = _gen_label_data(len(fov_names), len(comp_names), shape, dtype)
+
+    filelocs = {}
+
+    for i, fov in enumerate(fov_names):
+        tiffpath = os.path.join(base_dir, f'{fov}.tiff')
+        io.imsave(tiffpath, label_data[i, :, :, 0], plugin='tifffile')
+        filelocs[fov] = tiffpath
+
+    return filelocs, label_data
+
+
+TIFFMAKERS = {
+    'tiff': _create_tifs,
+    'multitiff': _create_multitiff,
+    'reverse_multitiff': _create_reverse_multitiff,
+    'mibitiff': _create_mibitiff,
+    'labels': _create_labels,
+}
+
+
+def create_paired_xarray_fovs(base_dir, fov_names, channel_names, img_shape=(1024, 1024),
+                              mode='tiff', delimiter=None, sub_dir=None, fills=False,
+                              dtype="int8"):
+
+    if not os.path.isdir(base_dir):
+        raise FileNotFoundError(f'{base_dir} is not a directory')
+
+    if fov_names is None or fov_names is []:
+        raise ValueError('No fov names were given...')
+
+    if channel_names is None or channel_names is []:
+        raise ValueError('No image names were given...')
+
+    if not isinstance(fov_names, list):
+        fov_names = [fov_names]
+
+    if not isinstance(channel_names, list):
+        channel_names = [channel_names]
+
+    filelocs, tif_data = TIFFMAKERS[mode](base_dir, fov_names, channel_names, img_shape, sub_dir,
+                                          fills, dtype)
+
+    if delimiter is not None:
+        fov_ids = [fov.split(delimiter)[0] for fov in fov_names]
+    else:
+        fov_ids = fov_names
+
+    if 'multitiff' in mode:
+        channel_names = range(len(channel_names))
+
+    if mode == 'labels':
+        data_xr = make_labels_xarray(tif_data, fov_ids, *img_shape, channel_names)
+    else:
+        data_xr = make_images_xarray(tif_data, fov_ids, *img_shape, channel_names)
+
+    return filelocs, data_xr
+
+
+def xarrays_are_equal(a_xr, b_xr, sortdims=("fovs", "channels")):
+    try:
+        if len(sortdims) > 1:
+            return xarrays_are_equal(
+                a_xr.sortby(sortdims[0]),
+                b_xr.sortby(sortdims[0]),
+                sortdims[1:]
+            )
+        else:
+            return b_xr.sortby(sortdims[0]).equals(a_xr.sortby(sortdims[0]))
+    except TypeError:
+        raise TypeError(f'sortdims argument must be an iterable. '
+                        f'sortdims given was, {sortdims}, which is of type, {type(sortdims)}')
+
+
+def make_images_xarray(tif_data, fov_ids, row_size, col_size, channel_names, dtype='int16'):
+    if tif_data is None:
+        tif_data = _gen_tif_data(len(fov_ids), len(channel_names), (row_size, col_size), False,
+                                 dtype)
+    coords = [fov_ids, range(row_size), range(col_size), channel_names]
+    dims = ["fovs", "rows", "cols", "channels"]
+    return xr.DataArray(tif_data, coords=coords, dims=dims)
+
+
+def make_labels_xarray(label_data, fov_ids, row_size, col_size, compartment_names):
```

Comment:
If we're assuming this is always 4D, we don't need to pass in row_size and col_size, we can get that from label_data

Reply:
I wasn't sure what to do there because of the `label_data=None` case, which produces synthetic label data.  I could maybe just set default values for row_size, col_size and use those if `label_data` is None, and otherwise get them from `label_data`'s shape.

Reply:
Won't this function fail with label_data=None?

Reply:
oh whoops, I was supposed to have that be a thing like in `make_images_xarray`